cudapoa/src/allocate_block.cpp

```diff
@@ -120,7 +120,13 @@ std::tuple<int64_t, int64_t, int64_t, int64_t> BatchBlock::calculate_space_per_p
     device_size_per_poa += (output_mask_ & OutputType::msa) ? poa_count * max_sequences_per_poa_ * sizeof(uint16_t) : 0; // input_details_d_->sequence_begin_nodes_ids
 
     // for graph - host
-    host_size_fixed += sizeof(GraphDetails); // graph_details_d_
+    host_size_fixed += sizeof(GraphDetails);                                                            // graph_details_h_
+    host_size_fixed += sizeof(GraphDetails);                                                            // graph_details_d_
```

Comment:
shouldn't this line move further down to the `// for graph - device` section?

Reply:
The device allocation for graph is already done. This is the host memory allocation for when parts of the graph are copied back

Reply:
ah sorry, yes, you are right. I misread this line. All good.